### ISSUE TITLE
Dotnet install doesn't fail when installing two tools with different package ids but the same Command Name

### ISSUE THREAD

**Issue by mattscheffer** [dotnet/cli#8479]
## Steps to reproduce
Create 2 projects to be used with different names
```
dotnet new console -n proj1
dotnet new console -n proj2
```
Edit the project file for each and add the following
```
<PackAsTool>True</PackAsTool>
<ToolCommandName>mytool1</ToolCommandName>
```
Should be the same command name for each.
Run dotnet pack in each project file folder
Copy the package to your local NuGet feed (or run the next commands with --source tags)
Attempt to install both
```
dotnet install tool -g proj1
dotnet install tool -g proj2
```

## Expected  behavior
The second install should fail as it has the same tool name as proj1.

## Actual behavior
Both install.
![image](https://user-images.githubusercontent.com/10537189/35416084-bf348000-01dc-11e8-9660-eb1cd6539280.png)

## Environment data
`dotnet --info` output:

.NET Command Line Tools (2.1.300-preview2-008017)

Product Information:
 Version:            2.1.300-preview2-008017
 Commit SHA-1 hash:  f4a0d2db0e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300-preview2-008017\

Microsoft .NET Core Shared Framework Host

  Version  : 2.1.0-preview2-26123-02
  Build    : b64cb46124df1f7e64d7b7cde1034f693b270ceb

**Comment by wli3** [comment]
@peterhuene did you work on stage-commit(rename) for tools yet? If not, I want take this change to implement that

**Comment by peterhuene** [comment]
I haven't worked on it yet, so all yours.

**Comment by wli3** [comment]
Fixed by #8496, and manually verified

**Comment by livarcocc** [comment]
@wli3 can we close this one then?
